core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/MetadataService.java

```diff
@@ -136,6 +139,28 @@ public MetadataTempFilesListDto uploadMetadataFilesToTempDir( InputStream metada
     return dto;
   }
 
+  protected InputStream extractXmiFile( InputStream metadataFile, FormDataContentDisposition schemaFileInfo ) throws Exception {
+    String fileNameMain = schemaFileInfo.getFileName();
```

Comment:
It might not be possible, but I would check that `schemaFileInfo` isn't null before calling a function.

Who knows in future upgrades of libraries, or with refactoring later with other callers of this function, it might be possible to get a null `schemaFileInfo`

We get so many jira issue where code doesn't check for null.